js/src/javascript/tokenizer.js

```diff
@@ -274,6 +274,12 @@ Tokenizer.prototype._read_non_javascript = function(c) {
         this._input.next();
       }
       return this._create_token(TOKEN.WORD, sharp);
+    } else if (this._input.hasNext()) {
+      // Handle private field (ie. #privateExample)
+      resulting_string = this.__patterns.identifier.read();
```

Comment:
Maybe we could add `#` to the `__patterns.identifier` as a valid start character?  I understand it isn't valid outside of classes, but it also isn't used for other things right?

Reply:
Yes, it was the shebang tests which broke. I'm not sure why they failed since there is specific handling for shebangs which changing the identifier start character shouldn't really have effected.

Reply:
The problem is here in the tokenizer: 
https://github.com/beautify-web/js-beautify/blob/c26968ab88761ca1b9cede3844cb52d3d9af03df/js/src/javascript/tokenizer.js#L166-L174

The non-javascript is parsed *after* javascript words.  So, matching for word that start with `#` takes precedence over shebang.  😢 

Please go back and implement this using `__patterns.identifier`.  The try moving the non-javascript detection to before word detection.  Then let's see what errors come up.  Okay?

Reply:
Thanks for pointing me in the right direction. Making it evaluate the non-JS stuff first works now with `#` added as a start to an identifier.